src/test/java/com/thebois/models/beings/BeingTests.java

```diff
@@ -249,4 +276,255 @@ public void equalsReturnsFalseForNotEqualBeings(
+        final Pawn pawn = new Pawn(
+            position,
+            position,
+            Mockito.mock(Random.class),
+            pathFinder,
+            mockFinder);
+
+        // Act
+        final float initialBuildStatus = structure.builtStatus();
+        pawn.update();
+        final float finalBuildStatus = structure.builtStatus();
+
+        // Assert
+
+        assertThat(initialBuildStatus).isLessThanOrEqualTo(finalBuildStatus);
+    }
+
+    @Test
+    public void pawnDoNotBuildsHouses() {
+        // Arrange
+        final IFinder mockFinder = Mockito.mock(IFinder.class);
+        final IPathFinder pathFinder = Mockito.mock(IPathFinder.class);
+        final Position position = new Position(0, 0);
+        final Position farAway = new Position(50, 50);
+        final IStructure structure = new House(farAway);
+        when(mockFinder.findNearestStructure(any())).thenReturn(Optional.of(structure));
+        final Pawn pawn = new Pawn(
+            position,
+            position,
+            Mockito.mock(Random.class),
+            pathFinder,
+            mockFinder);
+
+        // Act
+        final float initialBuildStatus = structure.builtStatus();
+        pawn.update();
+        final float finalBuildStatus = structure.builtStatus();
+
+        // Assert
+        // Value did not change because Pawn was not close enough
+        assertThat(initialBuildStatus).isEqualTo(finalBuildStatus);
+    }
+
+    @Test
+    public void doesPawnWalkToNearestUnBuiltBuilding() {
+        // Arrange
+        final Position positionA = new Position(10, 0);
+        final Position startPosition = new Position(20, 0);
+        final Position positionB = new Position(30, 0);
+
+        final Position correctDestination = new Position(11, 0);
+
+        final World world = new World(50, 0);
+        final IPathFinder pathFinder = new AstarPathFinder(world);
+
+        world.createStructure(positionA);
+        world.createStructure(positionB);
+
+        for (int i = 0; i < 10; i++) {
+            world.getStructureAt(positionB).get().deliverItem(new Rock());
+            world.getStructureAt(positionB).get().deliverItem(new Log());
+
+        }
+
+        final Pawn testPawn = new Pawn(
+            startPosition,
+            new Position(),
+            new Random(),
+            pathFinder,
+            world);
+
+        // Act
+        testPawn.update();
+
+        // Assert
+        assertThat(testPawn.getFinalDestination().get()).isEqualTo(correctDestination);
+    }
+
+    @Test
+    public void allStructuresCompleted() {
+        // Arrange
+        final Position positionA = new Position(10, 0);
+        final Position startPosition = new Position(20, 0);
+        final Position positionB = new Position(30, 0);
+
+        final Position correctDestination = new Position(0, 0);
+
+        final World world = new World(50, 0);
+        final IPathFinder pathFinder = new AstarPathFinder(world);
```

Comment:
If you don't need the functionality of `AstarPathFinder` then you should mock it. Same goes for world, but in this case it looks like it is required.